cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -122,4 +253,372 @@ private String getProgramScheduleId(Id.Program program, SchedulableProgramType p
     return String.format("%s:%s:%s:%s", program.getNamespaceId(), program.getApplicationId(),
                          programType.name(), program.getId());
   }
+
+  /**
+   * One instance of this class contains a list of {@link StreamSizeSchedule}s, which are all interested
+   * in the same stream. This instance subscribes to the size notification of the stream, and polls the
+   * stream for its size whenever the schedules it references need the information.
+   * The {@link StreamSizeScheduler} communicates with this class, which in turn communicates to the schedules
+   * it contains to perform operations on the schedules - suspend, resume, etc.
+   */
+  private final class StreamSubscriber implements NotificationHandler<StreamSizeNotification>, Cancellable {
```

Comment:
It doesn't have to be, but since this is very internal to the ``StreamSizeScheduler``... In any case, if I have to move it, I'll move it only after you reviewed my changes completely, since it would make it hard to see what's changed.